Crane/app/src/main/java/androidx/compose/samples/crane/base/ExploreSection.kt

```diff
@@ -93,21 +94,27 @@ private fun ExploreItem(
             .padding(top = 12.dp, bottom = 12.dp)
     ) {
         ExploreImageContainer {
-            CoilImage(
-                data = item.imageUrl,
-                fadeIn = true,
-                contentScale = ContentScale.Crop,
-                contentDescription = null,
-                loading = {
-                    Box(Modifier.fillMaxSize()) {
-                        Image(
-                            modifier = Modifier.size(36.dp).align(Alignment.Center),
-                            painter = painterResource(id = R.drawable.ic_crane_logo),
-                            contentDescription = null
-                        )
-                    }
+            Box {
+                val painter = rememberCoilPainter(
+                    request = item.imageUrl,
+                    fadeIn = true,
+                )
+                Image(
+                    painter = painter,
+                    contentDescription = null,
+                    contentScale = ContentScale.Crop,
+                )
+
+                if (painter.loadState == ImageLoadState.Loading) {
+                    Image(
+                        painter = painterResource(id = R.drawable.ic_crane_logo),
+                        contentDescription = null,
+                        modifier = Modifier
```

Comment:
Missing the `Modifier.fillMaxSize` that was previously in the `Box` of `loading`. Is it not needed anymore?

Reply:
It's makes no difference really, as a `ExploreImageContainer` is set to 60.dp, and the images are all larger than that... but I added an explicit `fillMaxSize()` anyway